conjure/conjure_test.go

```diff
@@ -1103,6 +1123,20 @@ func (c *testServiceClient) StreamResponse(ctx context.Context, authHeader beare
 	return resp.Body, nil
 }
 
+func (c *testServiceClient) MaybeStreamResponse(ctx context.Context, authHeader bearertoken.Token) (*io.ReadCloser, error) {
+	var requestParams []httpclient.RequestParam
+	requestParams = append(requestParams, httpclient.WithRPCMethodName("MaybeStreamResponse"))
+	requestParams = append(requestParams, httpclient.WithRequestMethod("GET"))
+	requestParams = append(requestParams, httpclient.WithHeader("Authorization", fmt.Sprint("Bearer ", authHeader)))
+	requestParams = append(requestParams, httpclient.WithPathf("/catalog/maybe/streamResponse"))
+	requestParams = append(requestParams, httpclient.WithRawResponseBody())
+	resp, err := c.client.Do(ctx, requestParams...)
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, nil
```

Comment:
Since this is the client code, I think the logic we actually want is:

```
if resp.StatusCode == http.StatusNoContent {
        return nil, nil
}
```

Then the contract for these functions is that a return value of `nil` for the `ReadCloser` indicates that there was no content (pretty sure that if the response is empty, then `resp.Body` will be non-nil but will return 0 bytes when read, which is the correct behavior)